gr-blocks/lib/CMakeLists.txt

```diff
@@ -168,15 +168,29 @@ add_library(gnuradio-blocks
     vector_map_impl.cc
     vector_to_stream_impl.cc
     vector_to_streams_impl.cc
-    wavfile_sink_impl.cc
-    wavfile_source_impl.cc
     )
 
-target_link_libraries(gnuradio-blocks
+if (SNDFILE_FOUND)
+    set(BLOCKS_SOURCES ${BLOCKS_SOURCES}
+        wavfile_sink_impl.cc
+        wavfile_source_impl.cc
+    )
+endif()
+
+add_library(gnuradio-blocks ${BLOCKS_SOURCES})
+
+set(BLOCKS_LIBS
     gnuradio-runtime
     Volk::volk
-    sndfile::sndfile
-)
+    )
+
+if (SNDFILE_FOUND)
+    set(BLOCKS_LIBS ${BLOCKS_LIBS}
+        sndfile
```

Comment:
Using the library name `sndfile` instead of the CMake target `sndfile::sndfile` seems to have come up because of issues with gr-osmosdr according to https://lists.gnu.org/archive/html/discuss-gnuradio/2020-09/msg00055.html. This fix feels like a workaround to another issue, either the target not being properly defined in CMake for OOT modules or that sndfile should be linked privately. As someone who has mucked around with packaging GR for conda, I much prefer a CMake error for a missing target to a linker error, or even worse, silently linking to a library that's not explicitly listed as a dependency (which it seems is what would happen currently with gr-osmosdr).

Reply:
It's over my head. @noc0lour Any comment?

Reply:
I was able to figure it out by using the FFT component as a guide. The primary issue was not adding SNDFILE to GR_LIBRARY_FOO in gr-blocks/lib/CMakeLists.txt.

Reply:
Update pushed.